### ISSUE TITLE
nix log <attribute> fails trying to execute "bro" ???

### ISSUE THREAD

**Issue by peti** [NixOS/nix#2003]
~~~
$ nix log nixpkgs.pandoc  | cat
error: executing 'bro': No such file or directory
[0.0 MiB DL]
error: writing to file: Broken pipe
~~~

I'm using:

system: "x86_64-linux", multi-user?: yes, version: nix-env (Nix) 2.0, channels(simons): "nixpkgs-17.09.1502.19879836d1", channels(root): "nixos-18.03pre130932.cc4677c36ee", nixpkgs: /home/simons/.nix-defexpr

**Comment by dtzWill** [comment]
This happens when brotli support was not available at compile time.

(Is this Nix 2 from 17.09?)

Just checked, looks like a nixpkgs "bug" that was fixed but not backported.

Commit: https://github.com/NixOS/nixpkgs/commit/7dafa09ed93e8fb932870d839b0668c9ed1c92ee

**Comment by peti** [comment]
This is Nix from the current `nixos-unstable` channel.

**Comment by dtzWill** [comment]
I would ask you to share the "nix log" for your copy of nix, but... lol.

What's the store path for your nix? Curious how we managed to not build with brotli support :3.

**Comment by peti** [comment]
~~~
❯ cd .nix-defexpr
❯ g reset --hard channel/nixos-unstable
HEAD is now at cc4677c36ee libcdio082: remove
❯ glg -1 | cat
commit cc4677c36ee8d880e881459ad114fd2224b3ac1c
Author: Piotr Bogdan <ppbogdan@gmail.com>
Date:   Sun Mar 4 18:31:06 2018 +0000

    libcdio082: remove

 pkgs/development/libraries/libcdio/0.82.nix | 30 -----------------------------
 pkgs/top-level/all-packages.nix             |  1 -
 2 files changed, 31 deletions(-)
❯ nixpkgs-build -A nix
/nix/store/hd04jgb1iyvps8w9c0iims3j8niyqdf0-nix-2.0
~~~~

**Comment by dtzWill** [comment]
```console
$ nix run -i /nix/store/hd04jgb1iyvps8w9c0iims3j8niyqdf0-nix-2.0 -c nix log /nix/store/hd04jgb1iyvps8w9c0iims3j8niyqdf0-nix-2.0

error: executing '': No such file or directory
downloading 'https://cache.nixos.org/log/h7yxc1h4f1frb78i5fx55hvv4d01k7wg-nix-2.0.drv'error: executing 'bro': No such file or directory
[0.0 MiB DL]
error: writing to file: Broken pipe
```

Hooray for being able to reproduce!

**Comment by dtzWill** [comment]
```console
$ nix show-derivation /nix/store/hd04jgb1iyvps8w9c0iims3j8niyqdf0-nix-2.0|& grep brotli
$
```

:(

**Comment by dtzWill** [comment]
Well, FWIW unstable-small has a fixed `nix`.

**Comment by dtzWill** [comment]
This is fixed on nixos-unstable--just checked with:

```console
$ nix run -f channel:nixos-unstable nix -c nix log /nix/store/hd04jgb1iyvps8w9c0iims3j8niyqdf0-nix-2.0
```

@peti please check if this resolves the problem for you! :)

**Comment by edolstra** [comment]
I also pushed the fix to 17.09.
